### ISSUE TITLE
Rule to ensure the local variables in function are camelCased

### ISSUE THREAD

**Issue by pushpa1981** [eslint/eslint#11050]
**What rule do you want to change?**
https://eslint.org/docs/rules/camelcase

**Does this change cause the rule to produce more or fewer warnings?**
More
**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**
 I am planning to setup rules on my projects  to make sure of all the local variables in a function should be camelCased. I didn't find a rule in eslint which supports this scenario hence requesting rule support for this.

```js
function(value)
{
     const NEW_VALUE=value;   
}
```

**What does the rule currently do for this code?**
nothing

**What will the rule do after it's changed?**
will error if there are any instances which doesn't have local variables camelCased.

**Are you willing to submit a pull request to implement this change?**
No.

**Comment by platinumazure** [comment]
Hi @pushpa1981, thanks for the issue.

The camelcase rule specifically allows for constant-case (all caps with underscores) as exceptions. (`ALL_CAPS` is allowed, `all_lower` and `Some_Lower`are not allowed.)

You can use the `id-match` rule to enforce the desired variable naming style without the all-caps exception.

**Comment by pushpa1981** [comment]
Thanks for suggestion @platinumazure . I have tried `id-match` rule and I see that this rule being applied to `var, function, and class ` but my scenario is to apply function and it's local variables. Is there any way i can exclude the class declarations for this rule? I appreciate any suggestions on this. BTW, I couldn't find the way to exclude the class declaration.
